gen_cert.py

```diff
@@ -229,7 +217,12 @@ def interstitial_factory():
             self.long_course = long_course or cert_data.get('LONG_COURSE', '').encode('utf-8')
             self.issued_date = issued_date or cert_data.get('ISSUED_DATE', '').encode('utf-8') or 'ROLLING'
             self.interstitial_texts = collections.defaultdict(interstitial_factory())
-            self.interstitial_texts.update(cert_data.get('interstitial', {}))
+            interstitial_dict = {k.encode('utf8'): v.encode('utf8') for k, v in cert_data.get('interstitial', {}).items()}
+            self.interstitial_texts.update(interstitial_dict)
+            self.locale = cert_data.get('LOCALE', settings.DEFAULT_LOCALE).encode('utf-8')
+            self.date_format = cert_data.get('DATE_FORMAT', settings.DEFAULT_DATE_FORMAT).encode('utf-8')
+            self.course_translations = cert_data.get('translations', '')
+            self.default_translations = settings.DEFAULT_TRANSLATIONS
```

Comment:
Do we need an instance-level reference to this? Shouldn't it still be accessible via `settings`?